frontend/packages/knative-plugin/src/topology/components/knativeComponentUtils.ts

```diff
@@ -46,24 +45,52 @@ export const canDropPubSubSinkOnNode = (operation: string, edge: Edge, node: Nod
   operation === MOVE_PUB_SUB_CONNECTOR_OPERATION &&
   !node.getTargetEdges().find((e) => e.getSource() === edge.getSource());
 
+export const isEventPubSubDroppable = (source: Node, target: Node) => {
+  return (
+    source.getType() === TYPE_EVENT_PUB_SUB &&
+    !source
+      .getSourceEdges()
+      .map((e) => e.getTarget())
+      .includes(target)
+  );
+};
+const getKnativeTooltip = (monitor): string => {
+  return monitor.getOperation()?.type === CREATE_PUB_SUB_CONNECTOR_OPERATION
+    ? monitor.getItem()?.getData()?.resources.obj.kind === EventingBrokerModel.kind
+      ? 'Add  Trigger'
+      : 'Add Subscription'
+    : 'Move sink to service';
```

Comment:
can we avoid one ternary and have 

```suggestion
 const sinkKind =  monitor.getItem()?.getData()?.resources.obj.kind;
  return monitor.getOperation()?.type === CREATE_PUB_SUB_CONNECTOR_OPERATION
    ?  `Add ${sinkKind}` : 'Move sink to service';
```

Reply:
While trying this i realised  `monitor.getItem()` is returning broker/channel during this operation, so we cant use this way.

Reply:
ah yes !!